src/main/kotlin/com/exactpro/th2/codec/util/ErrorMessageUtil.kt

```diff
@@ -17,35 +17,45 @@
 package com.exactpro.th2.codec.util
 
 import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.MessageGroupBatch
 import com.exactpro.th2.common.grpc.MessageMetadata
 import com.exactpro.th2.common.message.message
 import com.exactpro.th2.common.message.plusAssign
 import com.exactpro.th2.common.message.set
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.GroupBatch
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.MessageGroup
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.ParsedMessage
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage
+import com.exactpro.th2.common.utils.message.id
 import com.exactpro.th2.common.grpc.RawMessage as ProtoRawMessage
 import com.exactpro.th2.common.grpc.MessageGroup as ProtoMessageGroup
 
 const val ERROR_TYPE_MESSAGE = "th2-codec-error"
 const val ERROR_CONTENT_FIELD = "content"
 const val ERROR_EVENT_ID = "error_event_id"
 
-fun RawMessage.toErrorMessage(protocols: Collection<String>, errorEventId: EventID, errorMessage: String) =
-    ParsedMessage(
-        id,
-        eventId,
-        ERROR_TYPE_MESSAGE,
-        metadata,
-        protocol.ifBlank(protocols::singleOrNull) ?: protocols.toString(),
-        hashMapOf(ERROR_CONTENT_FIELD to errorMessage, ERROR_EVENT_ID to errorEventId.toString())
-    )
+fun RawMessage.toErrorMessage(
+    protocols: Collection<String>,
+    errorEventId: EventID?,
+    errorMessage: String
+): ParsedMessage = ParsedMessage(
+    id,
+    eventId,
+    ERROR_TYPE_MESSAGE,
+    metadata,
+    protocol.ifBlank(protocols::singleOrNull) ?: protocols.toString(),
+    hashMapOf(ERROR_CONTENT_FIELD to errorMessage).apply {
+        // Maybe use cradleString property instead of toString method.
+        errorEventId?.let { put(ERROR_EVENT_ID, errorEventId.toString()) }
```

Comment:
Since book and pages - yes, we should not use `toString()` method here because there is no use of it (the result value cannot be used anywhere without additional actions). We either should compute the string in the format that our providers understand. So, yep, we should try to use `cradleString` extension property here